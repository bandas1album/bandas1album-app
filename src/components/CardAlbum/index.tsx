const CardAlbum = () => (
  <div>
    <h1>teste</h1>
  </div>
)

export default CardAlbum
