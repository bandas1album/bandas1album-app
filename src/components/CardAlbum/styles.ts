import Link from 'next/link'
import styled from 'styled-components'

export const Card = styled.article`
  cursor: pointer;
  position: relative;
  overflow: hidden;
  width: 100%;
  height: 100%;
  border-radius: 8px;
  background-color: var(--color-primary-shade);

  &::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: block;
    background-color: transparent;
    transition: 0.2s background-color ease;
  }

  &:hover::after {
    background-color: rgba(0, 0, 0, 0.25);
  }
`

export const CardLink = styled(Link)`
  text-decoration: none;
`

export const CardImage = styled.figure`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  aspect-ratio: 1/1;
  margin: 0;
  object-fit: cover;

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`

export const CardTitle = styled.figcaption`
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: 8px;
  padding: 16px;
  text-align: center;
  text-decoration: none;
  color: var(--color-light);

  strong {
    font-size: 1rem;
  }

  span {
    font-size: 0.75rem;
  }
`
