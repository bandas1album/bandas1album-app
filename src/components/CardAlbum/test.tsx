import { render, screen } from '@testing-library/react'

import CardAlbum from '.'

describe('<CardAlbum />', () => {
  it('should render the heading', () => {
    const { container } = render(<CardAlbum />)

    expect(screen.getByRole('heading', { name: /test/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
