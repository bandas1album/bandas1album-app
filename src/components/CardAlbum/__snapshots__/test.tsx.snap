// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardAlbum /> should render the heading 1`] = `
<div>
  <h1>
    teste
  </h1>
</div>
`
